src/components/Select.tsx

```diff
@@ -7,40 +9,91 @@ interface SelectProps {
   label?: string;
   options?: Option[];
   value?: number | string;
-  onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+  onChange: (value: any) => void;
+  showSearchInput?: boolean;
+  showAllOption?: boolean;
 }
 
-function Select({label, options, value, onChange}: SelectProps) {
+function Select({
+  label,
+  options,
+  value,
+  onChange,
+  showSearchInput = false,
+  showAllOption = true,
+}: SelectProps) {
+  const currentOption = options?.find((option) => option.value == value);
+  const [searchText, setSearchText] = useState('');
+
   return (
     <div className="form-group" style={{width: '100%'}}>
       <label className="form-label" htmlFor="select">
         {label}
       </label>
-      <div style={{borderRadius: '20px'}} className="shadow-sm">
-        <select
-          style={{borderRadius: '20px'}}
-          className="form-select"
-          name="select"
-          id="select"
-          value={value}
-          onChange={onChange}
-        >
-          <option value="0">All</option>
-          {options &&
-            options
-              .sort((option1, option2) =>
-                option1.displayName < option2.displayName
-                  ? -1
-                  : option1.displayName > option2.displayName
-                  ? 1
-                  : 0
-              )
-              .map((option) => (
-                <option key={option.value} value={option.value}>
-                  {option.displayName}
-                </option>
-              ))}
-        </select>
+      <div className="d-flex">
+        <div className="dropdown mr-1 w-100">
+          <button
+            type="button"
+            className="btn btn-secondary dropdown-toggle w-100 text-left shadow-sm rounded-pill border-none text-start"
+            id="dropdown"
+            data-bs-toggle="dropdown"
+            aria-haspopup="true"
+            aria-expanded="false"
+            data-offset="10,20"
+          >
+            {currentOption?.displayName || 'All'}
+          </button>
+          <div
+            className="dropdown-menu overflow-scroll w-100 pt-0"
```

Comment:
I think overflow-auto is better here. Will independently do x/y scrollbars as needed.